### DISCUSSION TITLE
[Question] how to customize the replace range when completion item was selected?

### DISCUSSION THREAD

**Discussion by KamiC6238** [microsoft/monaco-editor#3676] [Q&A]
### Description
follow the image, the 'next' completion item actually is '->next', so when i pick the 'next' completion, it would be the second image
![image](https://user-images.githubusercontent.com/23523595/188559179-432203ac-e661-4925-8b2b-888db02c0545.png)

![image](https://user-images.githubusercontent.com/23523595/188559473-c6f7bf9e-4e46-4914-91fe-31d175d7afa2.png)

and i try to use 
`range: {
     insert: range,
     replace: range
}`

but not working, can anyone tell me how to cusomize the replace position to resolve that problem like the follow expected image?
![image](https://user-images.githubusercontent.com/23523595/188559953-56d04027-13a8-4bc5-a934-9e8a6adc0e0e.png)

**Comment by missfine** [comment]
How do you solve this problem now? I have the same problem. I also tried to use range to implement it. However, it is found that the effect is not consistent with the understanding of the document description, and the effect is very strange.

**Comment by KamiC6238** [comment]
> How do you solve this problem now? I have the same problem. I also tried to use range to implement it. However, it is found that the effect is not consistent with the understanding of the document description, and the effect is very strange.

hi, I finally solved it through additionalTextEdits,
![image](https://user-images.githubusercontent.com/23523595/212295314-33a21539-9482-4238-9f14-34bf2b16ef47.png)
but I only did this special treatment for c++ and c, and i hope this could help you.
